internal/apiserver/restartservice/restartimpl.go

```diff
@@ -63,6 +65,42 @@ func Restart(request *msgs.RestartRequest, pgouser string) msgs.RestartResponse
 		return resp
 	}
 
+	// if a rolling update is requested, this takes a detour to create a pgtask
+	// to accomplish this
+	if request.RollingUpdate {
+		// since a rolling update takes time, this needs to be performed as a
+		// separate task
+		// Create a pgtask
+		task := &crv1.Pgtask{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-%s", cluster.Name, config.LABEL_RESTART),
+				Namespace: cluster.Namespace,
+				Labels: map[string]string{
+					config.LABEL_PG_CLUSTER: cluster.Name,
+					config.LABEL_PGOUSER:    pgouser,
+				},
+			},
+			Spec: crv1.PgtaskSpec{
+				TaskType: crv1.PgtaskRollingUpdate,
+				Parameters: map[string]string{
+					config.LABEL_PG_CLUSTER: cluster.Name,
+				},
+			},
+		}
+
+		// remove any previous rolling restart, then add a new one
+		if err := apiserver.Clientset.CrunchydataV1().Pgtasks(task.Namespace).Delete(ctx, task.Name, metav1.DeleteOptions{}); err != nil && !kerrors.IsNotFound(err) {
+			log.Warnf("could not delete task: %s", err.Error())
```

Comment:
Well, it depends on _what_ errored. If it could not delete the pgtask because it did not exist, it would be created in the next step.

We're also attempting to do the creation in the next step anyway. If it was an error, we would exit out here.

Reply:
You're already handling `IsNotFound`, so it would be failing for another reason.

Reply:
That was a hypothetical. So let's say it's a transient error which, for whatever reason, we can't delete in that step but there is no existing CR, and we're able to create a step later.

Reply:
Anyway, we truly error if we try to create the pgtask and it cannot create.

Reply:
To provide an example, say the "delete" RBAC permission is missing for `pgtask`'s, which causes the delete to fail.  In this scenario, the user would see a "pgtask already exists" error, instead of the "missing RBAC" that truly caused it to fail.  This therefore hides the true root cause for the failure (the missing permission) from the error returned to the user, and requires looking at the operator logs for the specific warning (which also might not be easily identified via the logs being a warning and not an error).

And in general, I would consider the inability to delete a `pgtask` an error condition (vs a warning), since the operator cannot proceed if unable to do so.  Therefore, even within the logs (regardless of what is returned to the user) I would expect to see an error.

Reply:
Based on the nature of this API, I'm not completely convinced on this, but I have gone ahead and updated this to treat it as an error.